src/common/util/game.ts

```diff
@@ -513,15 +513,34 @@ export function dealDamageToObjectAtHex(state: w.GameState, amount: number, hex:
   return updateOrDeleteObjectAtHex(state, object, hex, cause);
 }
 
-export function updateOrDeleteObjectAtHex(state: w.GameState, object: w.Object, hex: w.HexId, cause: w.Cause | null = null): w.GameState {
+export function deleteAllDyingObjects(state: w.GameState): w.GameState {
+  Object.entries(allObjectsOnBoard(state)).forEach(([ hex, object ]) => {
```

Comment:
Any reason why this couldn't be a `.reduce` that returns a state with all update/delete actions applied?